lib/ansible/modules/files/copy.py

```diff
@@ -624,7 +626,7 @@ def main():
                     module.fail_json(msg="Destination directory %s is not accessible" % (os.path.dirname(dest)))
             module.fail_json(msg="Destination directory %s does not exist" % (os.path.dirname(dest)))
 
-    if not os.access(os.path.dirname(b_dest), os.W_OK) and not module.params['unsafe_writes']:
+    if not os.access(os.path.dirname(b_dest), os.W_OK) and not module.params['unsafe_writes'] and not module.check_mode:
```

Comment:
Similar to the logic changes requested above, this check should take into account the permissions of the directory in check mode if it exists. If it does not exist, check the nearest parent directory instead.